### PR TITLE
Support Pod Identity

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

1. Support Pod Identity
2. doc will be added after https://github.com/kurator-dev/kurator/pull/184

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: examples/aws-pod-identity/pod-indentity.yaml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        name: pod-identity-webhook
+        namespace: kube-system
+        path: "/mutate"
+    rules:
+      - operations: ["CREATE"]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["pods"]
+    sideEffects: None
+    admissionReviewVersions: ["v1beta1"]
```
Comment (hzxuzhonghu) [inline examples/aws-pod-identity/pod-indentity.yaml:86]: where does the yaml come from, use v1 here

Comment (hzxuzhonghu) [reply examples/aws-pod-identity/pod-indentity.yaml:86]: If this is all from aws provider, can we provide kubectl apply  <link>

File: docs/content/en/docs/cluster-operator/kurator-cluster-api.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+## Prerequisites
+
+{{< boilerplate prerequisites >}}
+
+## Build `cluster-operator` from source
+
+{{< boilerplate build-from-source >}}
+
+## Install cluster operator
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/cluster-operator/kurator-cluster-api.md:18]: We should have a separate guide for deploying cluster operator, then all the other examples will only need to refer it

File: docs/content/en/docs/cluster-operator/kurator-cluster-api.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+kubectl get cluster -w
+```
+
+Retrieve the cluster's Kubeconfig:
+
+```console
+clusterctl get kubeconfig quickstart > /root/.kube/quickstart.kubeconfig
+```
+
+Check node state:
```
Comment (hzxuzhonghu) [inline docs/content/en/docs/cluster-operator/kurator-cluster-api.md:44]: check what? We can depend on cluster status

File: pkg/apis/cluster/v1alpha1/cluster_consts.go [hunk 4]
```diff
@@ -37,6 +37,8 @@ const (
 
 	// ReadyCondition defines the Ready condition type that summarizes the operational state of a Cluster.
 	ReadyCondition capiv1.ConditionType = "Ready"
+	// ProviderFailedReason (Severity=Error) documents that the cluster infrastructure provider experienced an error.
+	ProviderFailedReason = "ProviderFailed"
```
Comment (hzxuzhonghu) [inline pkg/apis/cluster/v1alpha1/cluster_consts.go:41]: It is not explicit what does it mean, keep in mind all these are to endusers, they should be able to understand

File: pkg/infra/provider.go [hunk 5]
```diff
@@ -20,6 +20,8 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/aws/aws-sdk-go/aws"
```
Comment (hzxuzhonghu) [inline pkg/infra/provider.go:23]: nit: All these need to be refactored under aws

File: pkg/infra/provider.go [hunk 6]
```diff
@@ -52,20 +55,70 @@ type Provider interface {
...
+	if err := kube.Get(ctx, nn, secret); err != nil {
+		return nil, errors.Wrapf(err, "failed to get cluster secret %s", nn.String())
+	}
 
-type AWSProvider struct {
-	Kube  client.Client
-	scope *scope.Cluster
+	accessKeyID := string(secret.Data["AccessKeyID"])
+	secretAccessKey := string(secret.Data["SecretAccessKey"])
+	sessionToken := string(secret.Data["SessionToken"])
```
Comment (hzxuzhonghu) [inline pkg/infra/provider.go:104]: define three const

File: pkg/infra/openid/cert.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+	PrivateKey []byte
+	PublicKey  []byte
+	KeyResponse
+}
+
+type KeyResponse struct {
+	Keys []jose.JSONWebKey `json:"keys"`
+}
+
+func NewCert() (*Cert, error) {
```
Comment (hzxuzhonghu) [inline pkg/infra/openid/cert.go:42]: declare the original source

File: docs/content/en/docs/cluster-operator/kurator-cluster-api.md (hunk 2 above)
Comment (zirain) [reply docs/content/en/docs/cluster-operator/kurator-cluster-api.md:18]: follow up https://github.com/kurator-dev/kurator/issues/191

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**